docs/reference/api/connectivity/networksocket/TCPSocket.md

```diff
@@ -2,6 +2,20 @@
 
 The TCPSocket class provides the ability to send a stream of data over TCP. TCPSockets maintain a stateful connection that starts with the `connect` member function. After successfully connecting to a server, you can use the `send` and `recv` member functions to send and receive data (similar to writing or reading from a file).
 
+The constructor takes in the [NetworkStack](/docs/v5.6/mbed-os-api-doxy/class_network_stack.html) or [NetworkInterface](/docs/v5.6/mbed-os-api-doxy/class_network_interface.html) pointer to open the socket on specified NetworkInterface. If NetworkInterface is not passed in constructor, than you must call `open` to initialize the socket.
```

Comment:
I think it takes an implementation of the NetworkStack or Interface but why both? can we do without the links?

Reply:
than should be then

Reply:
Ugh. I can't believe I missed that twice. Thanks @sg-

Reply:
Ah. It's because it was after my copy edit. Phew. I feel better.